src/kernels/debugger/kernelDebugAdapterBase.ts

```diff
@@ -89,18 +90,9 @@ export abstract class KernelDebugAdapterBase implements DebugAdapter, IKernelDeb
             this.debugCell = notebookDocument.cellAt(configuration.__cellIndex!);
         }
 
+        this.jupyterSession.kernel?.iopubMessage.connect(this.onIOPubMessage, this);
         this.disposables.push(
-            this.jupyterSession.onIOPubMessage(async (msg: KernelMessage.IIOPubMessage) => {
```

Comment:
This is what I'd like to fix in this PR.
We need a way to hook up to the iopub and anymessage events in the underlyiing Jupyter kernel connection.
This property `onIOPubMessage` is a custom event we wire up. If I were to introduce anyMessage then i'd have to handle kernel restarts etc... As we expose the underlying kernel everywhere it makes sense to just hook to those events.

Hence created a kernel wrapper so that we have a single instance, of the Kernel.KernleConnection on which we can monitor the events.